### PR TITLE
Add names and namescopes to improve TensorBoard graph layout

### PR DESCRIPTION
**Pull request by wkeithvan**
The current TensorBoard graph is quite difficult to understand and very complicated:

ugly_tensorboard

By adding `tf.name_scope` around logically grouped operations (e.g. a down convolution or cross entropy operation), as well as by giving default names and adding optional name arguments to the variable creation functions in `tf_unet/layers.py`, the graph becomes much cleaner and more recognizable as a U-Net (note: keep_probability must be "removed from main graph" on TensorBoard to produce this exact visualization):

nice_tensorboard
-
There is no difference to the underlying operations, just a bit of cleanup to the names/name_scopes.

### CODE REVIEW

**Comment by jakeret** [comment]
This is great! Thanks a lot. I'll have a closer look

**Comment by wkeithvan** [comment]
One thing I just noticed...

In `unet.py` (line 204), the cross_entropy variable is assigned by
`self.cross_entropy = tf.reduce_mean(cross_entropy(tf.reshape(self.y, [-1, n_class]), tf.reshape(pixel_wise_softmax_2(logits), [-1, n_class])))`

which references the cross_entropy function in `layers.py`. In `layers.py` (line 71), cross_entropy is defined as
`return -tf.reduce_mean(y_*tf.log(tf.clip_by_value(output_map,1e-10,1.0)), name="cross_entropy")`

Is there a reason that the `tf.reduce_mean` function is called both in the function definition and on the result of the function? It seems to me that is just calling the function twice on itself...

If this is a mistake, another commit should be made to move the `tf.name_scope` call into the function definition (`layers.py`).

**Comment by jakeret** [comment]
This is brilliant! Great work, thank you very much.

I think you're right. The `reduce_mean` is incorrectly done twice.

Another thing: Could you add a name in the [deconv2d function](https://github.com/wkeithvan/tf_unet/blob/bfd9da2a6eb8acb286f78ff6a95ce5e5744f3662/tf_unet/layers.py#L39)? Without the up-conv node still look a bit confusing.

**Comment by wkeithvan** [comment]
Made the updates and checked it on the toy problem... nothing crashes, so that seems good. One problem is that I'm getting that issue with the toy problem predicting a black screen again :/

**Comment by wkeithvan** [comment]
Solved a minor issue with the xent namescope representation. This should be ready to merge, @jakeret
